sklearn/metrics/_pairwise_distances_reduction/_base.pyx.tp

```diff
@@ -73,7 +75,12 @@ cpdef DTYPE_t[::1] _sqeuclidean_row_norms32(
         )
 
     with nogil, parallel(num_threads=num_threads):
-        thread_num = openmp.omp_get_thread_num()
+
+        IF SKLEARN_OPENMP_PARALLELISM_ENABLED:
+            thread_num = openmp.omp_get_thread_num()
+        ELSE:
+            thread_num = 1
```

Comment:
This is a thread number, not a number of threads. So I think it should be 0 instead of 1.

But actually it would be better to rely on `sklearn.utils._openmp_helpers._openmp_thread_num` which is already cimported in this file instead.